sdk/compute/Azure.ResourceManager.Compute/src/Customize/SshPublicKeyResource.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Compute.Models;
+using Azure.ResourceManager.Resources;
+
+namespace Azure.ResourceManager.Compute
+{
+    public partial class SshPublicKeyResource
+    {
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public virtual async Task<Response<SshPublicKeyGenerateKeyPairResult>> GenerateKeyPairAsync(CancellationToken cancellationToken)
+        {
+            using var scope = _sshPublicKeyClientDiagnostics.CreateScope("SshPublicKeyResource.GenerateKeyPair");
+            scope.Start();
+            try
+            {
+                var response = await _sshPublicKeyRestClient.GenerateKeyPairAsync(Id.SubscriptionId, Id.ResourceGroupName, Id.Name, null,cancellationToken).ConfigureAwait(false);
+                return response;
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
```

Comment:
well the point of my previous comment is that we should call the new public version of this method, such as:
```suggestion
        public virtual async Task<Response<SshPublicKeyGenerateKeyPairResult>> GenerateKeyPairAsync(CancellationToken cancellationToken)
        => await GenerateKeyPairAsync(null, cancellationToken).ConfigureAwait(false);
```
so that this method will never require a change in the future

Reply:
this should apply to all methods with similar purpose.

Reply:
i've updated them accordingly. will you take a look?